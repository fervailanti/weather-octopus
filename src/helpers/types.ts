import { Dispatch } from 'react'
import { PinnedCityContext } from '../providers/PinnedCity'

export interface CoordType {
  lat: number | undefined
  lon: number | undefined
}

export interface TempType {
  current?: number
  min: number,
  max: number
}

export interface WeatherType {
  icon: string
  temp: TempType
  description?: string
}

export interface PinnedCityType {
  id: number,
  coord: CoordType,
  isLoading: boolean
}

export interface PinnedCityAction {
  type: 'id' | 'coord' | 'loading'
  id?: number
  coord?: CoordType
  loading?: boolean
}

export interface PinnedCityContextType {
  pinnedCity: PinnedCityType
  setPinnedCity: Dispatch<PinnedCityAction>
}

export interface CityType {
  id?: number
  name: string
  country: string
  time: string
  weather: WeatherType
}

export interface DayType {
  date: string
  weather: WeatherType
}

export type DailyForecastType = DayType[]
