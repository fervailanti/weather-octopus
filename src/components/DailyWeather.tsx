import React, { FC, ReactElement } from 'react'
import { DailyForecastType, DayType } from '../helpers/types'
import Day from './Day'

interface DailyWeatherProps {
  dailyForecast: DailyForecastType
}

const DailyWeather: FC<DailyWeatherProps> = ({ dailyForecast }): ReactElement => (
  <div className="flex justify-around h-full items-center flex-wrap mt-6 border-t">
    {dailyForecast.map((day: DayType, index: number) => (
      <Day date={day.date} weather={day.weather} key={index} />
    ))}
  </div>
)

export default DailyWeather
