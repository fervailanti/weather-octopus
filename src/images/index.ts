const req = (route: string) => require(`../images/gallery/${route}`)

export const images: any = {
  common: {
    octopus: req('common/octopus.png'),
    reload: req('common/reload.svg')
  },
  cities: {
    newYork: req('cities/new_york.jpg'),
    rome: req('cities/rome.jpg'),
    madrid: req('cities/madrid.jpg'),
    tokyo: req('cities/tokyo.jpg'),
    paris: req('cities/paris.jpg'),
    berlin: req('cities/berlin.jpg'),
  },
  weather: {
    icons: {
      sun: req('icons/sun.svg'),
      moon: req('icons/moon.svg'),
      sunCloudy: req('icons/sun-cloudy.svg'),
      moonCloudy: req('icons/moon-cloudy.svg'),
      clouds: req('icons/clouds.svg'),
      wind: req('icons/wind.svg'),
      rain: req('icons/rain.svg'),
      showerRain: req('icons/shower-rain.svg'),
      sunRain: req('icons/sun-rain.svg'),
      moonRain: req('icons/moon-rain.svg'),
      storm: req('icons/storm.svg'),
      snow: req('icons/snow.svg'),
      mist: req('icons/mist.svg'),
      forecast: req('icons/forecast.svg')
    }
  }
}