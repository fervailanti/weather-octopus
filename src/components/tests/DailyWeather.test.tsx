import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { images } from '../../images'
import DailyWeather from '../DailyWeather'
import { DailyForecastType } from '../../helpers/types'

const mockDailyForecast: DailyForecastType = Array(5).fill({
  date: 'Monday 20th',
  weather: {
    icon: images.weather.icons.forecast,
    temp: {
      min: 16,
      max: 19
    }
  }
})

describe('DailyWeather Component', () => {
  it('Should render correctly', () => {
    const daily = render(<DailyWeather dailyForecast={mockDailyForecast} />)
    expect(daily.container).toMatchSnapshot()
    expect(daily.container).toHaveTextContent('Monday 20th')
    expect(daily.container).toHaveTextContent('16ยบ / 19ยบ')
  })
})
