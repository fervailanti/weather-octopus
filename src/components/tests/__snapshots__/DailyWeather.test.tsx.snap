// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeather Component Should render correctly 1`] = `
<div>
  <div
    class="flex justify-around h-full items-center flex-wrap mt-6 border-t"
  >
    <div
      class="flex items-center mt-4 mr-4"
    >
      <img
        alt="weather-icon"
        class="h-6 mr-2"
        src="[object Object]"
      />
      <div>
        <p
          class="text-white font-thin text-sm"
        >
          Monday 20th
        </p>
        <p
          class="text-white font-thin text-sm"
        >
          <span
            class="font-normal"
          >
            16
          </span>
          º /
           
          <span
            class="font-normal"
          >
            19
          </span>
          º
        </p>
      </div>
    </div>
    <div
      class="flex items-center mt-4 mr-4"
    >
      <img
        alt="weather-icon"
        class="h-6 mr-2"
        src="[object Object]"
      />
      <div>
        <p
          class="text-white font-thin text-sm"
        >
          Monday 20th
        </p>
        <p
          class="text-white font-thin text-sm"
        >
          <span
            class="font-normal"
          >
            16
          </span>
          º /
           
          <span
            class="font-normal"
          >
            19
          </span>
          º
        </p>
      </div>
    </div>
    <div
      class="flex items-center mt-4 mr-4"
    >
      <img
        alt="weather-icon"
        class="h-6 mr-2"
        src="[object Object]"
      />
      <div>
        <p
          class="text-white font-thin text-sm"
        >
          Monday 20th
        </p>
        <p
          class="text-white font-thin text-sm"
        >
          <span
            class="font-normal"
          >
            16
          </span>
          º /
           
          <span
            class="font-normal"
          >
            19
          </span>
          º
        </p>
      </div>
    </div>
    <div
      class="flex items-center mt-4 mr-4"
    >
      <img
        alt="weather-icon"
        class="h-6 mr-2"
        src="[object Object]"
      />
      <div>
        <p
          class="text-white font-thin text-sm"
        >
          Monday 20th
        </p>
        <p
          class="text-white font-thin text-sm"
        >
          <span
            class="font-normal"
          >
            16
          </span>
          º /
           
          <span
            class="font-normal"
          >
            19
          </span>
          º
        </p>
      </div>
    </div>
    <div
      class="flex items-center mt-4 mr-4"
    >
      <img
        alt="weather-icon"
        class="h-6 mr-2"
        src="[object Object]"
      />
      <div>
        <p
          class="text-white font-thin text-sm"
        >
          Monday 20th
        </p>
        <p
          class="text-white font-thin text-sm"
        >
          <span
            class="font-normal"
          >
            16
          </span>
          º /
           
          <span
            class="font-normal"
          >
            19
          </span>
          º
        </p>
      </div>
    </div>
  </div>
</div>
`;
